docs/ecu_events.adoc

```diff
@@ -0,0 +1,177 @@
+
+Contains a correlation id.
+
+Example:
+
+```
+{
+        "deviceTime" : "2018-10-25T12:35:42Z",
+        "event" :
+        {
+                "correlationId" : "here-ota:mtus:c017f968-5c7e-4bf6-9992-2134ff06432c",
+                "ecu" : "CA:FE:A6:D2:84:9D"
+        },
+        "eventType" :
+        {
+                "id" : "EcuInstallationStarted",
+                "version" : 0
+        },
+        "id" : "9e3ad5f6-091d-4f63-96eb-20ba2e256631"
+}
+```
+
+=== EcuInstallationCompleted
+
+Emitted when installation on a specific ecu has ended (success or failure).
+
+Success/failure indicated by the boolean `event.success`.
+TODO: revisit that, maybe a status code instead + free form log string
+
+Some devices need a reboot after an installation to actually have an effect on the system.
+TODO: use a status code `NEED_REBOOT`?
+
+Contains a correlation id.
+
+Example:
+
+```
+{
+        "deviceTime" : "2018-10-25T12:35:42Z",
+        "event" :
+        {
+                "correlationId" : "here-ota:mtus:c017f968-5c7e-4bf6-9992-2134ff06432c",
+                "ecu" : "CA:FE:A6:D2:84:9D",
+                "success" : true
+        },
+        "eventType" :
+        {
+                "id" : "EcuInstallationCompleted",
+                "version" : 0
+        },
+        "id" : "74060fe8-c594-4f37-b676-4d3bedcb5ca5"
+}
+```
+
+== General reporting events
+
+=== DownloadComplete
+
+Emitted when all ecu images have been downloaded.
+
+TODO: Still useful?
+
+Example:
+
+```
+{
+        "deviceTime" : "2018-10-25T12:35:42Z",
+        "event" : "{\"signatures\":[{\"keyid\":\"798f4a19aa185f97c3c119e78f251cc30f293b226c3dc1ccd8694f15b1f36d5d\",\"method\":\"rsassa-pss\",\"sig\":\"ck5pu2LPxFYs39d6eDauGseBHWwmHTE6VNJ2Ngs+Iu6zveFjYPf/IU6P/ucBkWC8MnlgswLZeIznXU/+ynSoL95W5wHWYDxKruzTOCYUmXJxTFK+q4cnnfOgvBmoDh35M889vrqFQtsp2SG8P6VIe4iPnb02wbopVcW4FcCLks+3idg8lk15W0Pm3yedkmu7ycJk8ZuY11V8Xrt4at+DCcsK+TD0YiKmbypsBDAkS+lyjYz7aYqyXiLf0r4RbdIaBs6SM+KxcXb0QQK5wDEe0vbek5WYTz6fp91PED/q5KkrN60LCnIxRKSxg5dT06GJqcejr59r80OwOj0okYrqGw==\"}],\"signed\":{\"_type\":\"Targets\",\"custom\":{\"correlationId\":\"here-ota:mtus:c017f968-5c7e-4bf6-9992-2134ff06432c\"},\"expires\":\"2021-07-04T16:33:27Z\",\"targets\":{\"primary_firmware.txt\":{\"custom\":{\"ecuIdentifiers\":{\"CA:FE:A6:D2:84:9D\":{\"hardwareId\":\"primary_hw\"}}},\"hashes\":{\"sha256\":\"74e653bbf6c00a88b21f0992159b1002f5af38506e6d2fbc7eb9846711b2d75f\",\"sha512\":\"91814ad1c13ebe2af8d65044893633c4c3ce964edb8cb58b0f357406c255f7be94f42547e108b300346a42cd57662e4757b9d843b7acbc090df0bc05fe55297f\"},\"length\":59},\"secondary_firmware.txt\":{\"custom\":{\"ecuIdentifiers\":{\"secondary_ecu_serial\":{\"hardwareId\":\"secondary_hw\"}}},\"hashes\":{\"sha256\":\"1bbb15aa921ffffd5079567d630f43298dbe5e7cbc1b14e0ccdd6718fde28e47\",\"sha512\":\"7dbae4c36a2494b731a9239911d3085d53d3e400886edb4ae2b9b78f40bda446649e83ba2d81653f614cc66f5dd5d4dbd95afba854f148afbfae48d0ff4cc38a\"},\"length\":15}},\"version\":2}}",
```

Comment:
why is this one escaped?

Reply:
Yes, that's not great but that's what is going out now. Not sure if it's a regression or the historical behavior.

At a minimum, it could be:

```
"event" : {
  "targets": {
    "signatures": [ ... ]
    ...
  }
}
```

But I think @jerrytrieu said that we don't even care about this field.

Should I make the change anyway?

Reply:
That is historical behavior, for better or worse. That was the first event we created for this system and at the time Vladimir didn't seem to care about the details and I didn't put much thought into it. I'd say it is worth making it consistent with the other event reports, though.

Reply:
Yes, let's make this send `correlationId` as the others. Along with `ecu` and `target` filename.